pkg/apply/applier.go

```diff
@@ -47,9 +47,12 @@ func NewApplier(factory util.Factory, ioStreams genericclioptions.IOStreams) *Ap
 		factory:      factory,
 		ioStreams:    ioStreams,
 	}
-	a.previousInventoriesFunc = a.previousInventories
+	a.previousInventoriesFunc = a.PruneOptions.GetPreviousInventoryObjects
 	a.infoHelperFactoryFunc = a.infoHelperFactory
 	a.manifestReaderFunc = a.readObjects
+	a.InventoryFactoryFunc = prune.WrapInventoryObj
+	a.PruneOptions.InventoryFactoryFunc = prune.WrapInventoryObj
```

Comment:
Not sure I like that we end up setting these function properties in multiple places, but we can continue to improve as we add new implementations of the Inventory interface.